benchmark/algorithms/faiss_gpu.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import absolute_import
+#import sys
+#sys.path.append("install/lib-faiss")  # noqa
+import numpy
+import sklearn.preprocessing
+import ctypes
+import faiss
+import os
+import subprocess
+from benchmark.algorithms.base import BaseANN
+from benchmark.datasets import DATASETS
+from benchmark.sensors.power_capture import power_capture
+
+class FaissGPU(BaseANN):
+    def query(self, X, n):
+        if self._metric == 'angular':
+            X /= numpy.linalg.norm(X)
+        if not power_capture.enabled():
+            print("query size", X.size, X.shape) 
+        self.res = self.gindex.search(X.astype(numpy.float32, copy=False), n)
+        #GW self.res = self.gindex.search(X.view(numpy.float32), n)
+
+    def get_results(self):
+        D, I = self.res
+        return I
+#        res = []
+#        for i in range(len(D)):
+#            r = []
+#            for l, d in zip(L[i], D[i]):
+#                if l != -1:
+#                    r.append(l)
+#            res.append(r)
+#        return res
+
+
+class FaissIVFGPU(FaissGPU):
+    def __init__(self, metric, n_list):
+        self._n_list = n_list
+        self._metric = metric
+        self.index = None
+        self.gindex = None
+        self.quantizer = None
+        self.res = None
+        self.resource = None
+
+    def done(self):
+        if self.gindex:
+            self.gindex.reset() # release GPU and memory?
+
+    def index_name(self, name):
+        return f"data/ivf_{name}_{self._n_list}_{self._metric}"
+
+    def fit(self, dataset):
+        X = DATASETS[dataset].get_dataset() 
```

Comment:
The gpu variant is another baseline for in-memory datasets, right?

Reply:
This was my first attempt at a faiss gpu algorithm.  Obviously, we moved to the index factory - so I will remove this one.